sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/ShuffleExchangeExec.scala

```diff
@@ -225,7 +231,8 @@ object ShuffleExchangeExec {
       outputAttributes: Seq[Attribute],
       newPartitioning: Partitioning,
       serializer: Serializer,
-      writeMetrics: Map[String, SQLMetric])
+      writeMetrics: Map[String, SQLMetric],
+      isNullAwareAntiJoin: Boolean = false)
```

Comment:
I am wondering if this should be renamed to something like `countRowsWithNullPartitionKey` ? In the context of an exchange, having a variable that is very join specific is a bit weird, because join is but one of the things that causes an exchange.

Reply:
how about `collectNullPartitionKeyMetrics`

Reply:
Works for me !